src/pip/_internal/metadata/importlib/_envs.py

```diff
@@ -68,6 +68,8 @@ def _find_impl(self, location: str) -> Iterator[FoundResult]:
             info_location = get_info_location(dist)
             try:
                 name = get_dist_canonical_name(dist)
+                distribution = Distribution(dist, info_location, None)
+                getattr(distribution, "version")
```

Comment:
I don't understand why you've added this code, though. The original issue didn't suggest there was anything wrong here, and it's causing errors in cases that currently *aren't* raising an unhandled exception.

Is this simply an over-enthusiastic attempt to catch additional cases of bad data? If so, we'd normally avoid doing things like that, because backward compatibility is critical in a project that's used in as many places as pip. In particular, we have to be *exceptionally* careful to be forgiving in the case of bad data, because there's so much older, often legacy, code out there that we need to continue supporting.